testutil/race_on.go

```diff
@@ -0,0 +1,7 @@
+//go:build race
+
+package testutil
+
+const (
+	RaceEnabled = true
+)
```

Comment:
Review: this may be controversial, but conditionally skipping tests that we _know_ are flaky and slow under the race detector is preferable to just leaving these tests languishing in a perma-skipped state.